flintrock/core.py

```diff
@@ -506,42 +506,66 @@ def get_java_major_version(client: paramiko.client.SSHClient):
                 client=client,
                 command=command)
             tokens = output.split()
-            # First line of the output is like: 'java version "1.8.0_20"'
+            # First line of the output is like: 'openjdk version "1.8.0_252"' or 'openjdk version "11.0.7" 2020-04-14'
             # Get the version string and strip out the first two parts of the
             # version as a tuple: (1, 8)
             if len(tokens) >= 3:
                 version_parts = tokens[2].strip('"').split(".")
                 if len(version_parts) >= 2:
-                    return tuple(int(part) for part in version_parts[:2])
+                    if version_parts[0] == "1":
+                        # Java 6, 7 or 8
+                        return version_parts[1]
+                    else:
+                        # Java 9+
+                        return version_parts[0]
```

Comment:
Do we expect this method's return to always be an integer? If so, I think we should convert these strings to integers accordingly.

Actually, it looks like we are defining the JDK as an integer in the command-line interface, so I think we should match that here.